common/src/network.rs

```diff
@@ -3,12 +3,37 @@
 use bitcoin::blockdata::block::{Block, BlockHeader};
 use bitcoin::consensus::params::Params;
 use bitcoin::hash_types::BlockHash;
+use bitcoin::network::constants::ServiceFlags;
 use bitcoin_hashes::hex::FromHex;
 
 use bitcoin_hashes::sha256d;
 
 use crate::block::Height;
 
+/// Peer services supported by nakamoto.
+#[derive(Debug, Copy, Clone)]
+pub enum Services {
+    /// Peers with compact filter support
+    All,
+    /// Peers with only block support.
+    Chain,
+}
+
+impl From<Services> for ServiceFlags {
+    fn from(value: Services) -> Self {
+        match value {
+            Services::All => Self::COMPACT_FILTERS,
+            _ => Self::NETWORK,
```

Comment:
It's better to not use a catch-all here, because when a new variant is added, we'll forget to update this method.